src/basic/fd-util.c

```diff
@@ -464,6 +464,55 @@ int close_all_fds(const int except[], size_t n_except) {
         return r;
 }
 
+int pack_fds(int *fds, size_t n_fds) {
```

Comment:
btw, i'd write the first arg as `int fds[]`, to make clear it's an array we reference here not just a single value

but doesn#t matter...